File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
// Jai Ganesh
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int len = nums.size();
        unordered_map<int, int> lastSeen;
        for(int index = 0; index < len; index++) {
            if(lastSeen.find(target - nums[index]) != lastSeen.end()) {
                return {lastSeen[target - nums[index]], index};
            }
            lastSeen[nums[index]] = index;
        }

        return {-1,-1};
    }
};

File: Data Structures & Algorithms/missing-number/submission-0.java
// Jai Ganesh
class Solution {
    public int missingNumber(int[] nums) {
        int sum = 0;

        int expected = (nums.length * (nums.length + 1)) / 2;

        for(int ind = 0; ind < nums.length; ind++) {
            sum += nums[ind];
        }

        return expected - sum;
    }
}
